### PR TITLE
Fix devices reuse

### PR DESCRIPTION
**Pull request by imobachgs**
Fix several issues when trying to reuse devices in AutoYaST and improves error handling.

### CODE REVIEW

File: package/yast2-storage-ng.changes [hunk 1]
```diff
@@ -1,4 +1,16 @@
 -------------------------------------------------------------------
+Tue Oct 31 12:14:18 UTC 2017 - igonzalezsosa@suse.com
+
+- AutoYaST: fix reuse of partitions (bsc#1060637).
+- AutoYaST: improve error handling when trying to reuse a partition
+  fails.
+- AutoYaST: when a problem is detected while creating a proposal,
+  register in which section it was found.
+- AutoYaST: register an issue when a proposel does not contain
```
Comment (joseivanlopez) [inline package/yast2-storage-ng.changes]: Typo: proposel --> proposal

File: test/y2storage/proposal/autoinst_devices_planner_test.rb [hunk 2]
```diff
@@ -98,6 +98,42 @@
...
+
+        it "adds a new partition" do
+          devices = planner.planned_devices(drives_map)
+          root = devices.find { |d| d.mount_point == "/" }
+          expect(root.reuse).to be_nil
+        end
+
+        it "registers an issue" do
+          planner.planned_devices(drives_map)
+          issue = issues_list.find { |i| i.is_a?(Y2Storage::AutoinstIssues::MissingReusableDevice) }
```
Comment (joseivanlopez) [inline test/y2storage/proposal/autoinst_devices_planner_test.rb]: I would check that the issues list is empty or does not contain that issue before planning the devices.

File: test/y2storage/proposal/autoinst_devices_planner_test.rb [hunk 3]
```diff
@@ -98,6 +98,42 @@
...
+
+        it "adds a new partition" do
+          devices = planner.planned_devices(drives_map)
+          root = devices.find { |d| d.mount_point == "/" }
+          expect(root.reuse).to be_nil
+        end
+
+        it "registers an issue" do
+          planner.planned_devices(drives_map)
+          issue = issues_list.find { |i| i.is_a?(Y2Storage::AutoinstIssues::MissingReuseInfo) }
```
Comment (joseivanlopez) [inline test/y2storage/proposal/autoinst_devices_planner_test.rb]: Same comment than before. In general, when finding some issue, first I would check that the issue does not exist.

Comment (imobachgs) [reply test/y2storage/proposal/autoinst_devices_planner_test.rb]: Issues are supposed to be added when running the planner (`planned_services`), so I think it is safe to assume that there are no issues at the beginning of the test. But I've added them to be on the safe side :)

**Pull request merged**